apps/src/blockly/addons/cdoSerializationHelpers.ts

```diff
@@ -432,28 +439,27 @@ export function convertFunctionsXmlToJson(functionsXml: string) {
  * TODO: define a type for projectState and proceduresState. Blockly defines these as any.
  */
 export function appendProceduresToState(
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  projectState: any,
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  proceduresState: any
+  projectState: WorkspaceSerialization,
+  proceduresState: WorkspaceSerialization
 ) {
   const projectBlocks = projectState.blocks?.blocks || [];
   const projectProcedures = projectState.procedures || [];
 
-  // TODO: define these types better.
-  const sharedBlocks =
-    (proceduresState.blocks?.blocks as JsonBlockConfig[]) || [];
-  const sharedProcedures = (proceduresState.procedures as {id: string}[]) || [];
+  const sharedBlocks = proceduresState.blocks?.blocks || [];
+  const sharedProcedures = proceduresState.procedures || [];
 
   sharedBlocks.forEach(block => {
     const {behaviorId, procedureId} = block.extraState;
 
     if (!blockExists(behaviorId, projectBlocks)) {
       // If the block doesn't exist, add it to the student project
       projectBlocks.push(block);
-      projectProcedures.push(
-        sharedProcedures.find(procedure => procedure.id === procedureId)
+      const procedure = sharedProcedures.find(
+        procedure => procedure.id === procedureId
       );
+      if (procedure) {
+        projectProcedures.push(procedure);
+      }
```

Comment:
I think we're doing this because `.find()` doesn't guarantee a result that we'd want, even though in our case we'd expect to always find one.

Reply:
Can you clarify what you mean here? Is the new null check an issue?

Reply:
No I don't think so! I was just making sure I understood why we were adding it.

Reply:
Yeah `find` could theoretically find nothing.